huffman/src/detail/flattened_symbol_bitsize_view.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include "huffman/src/detail/iterator_interface.hpp"
+
+#include <cstddef>
+#include <iterator>
+#include <numeric>
+#include <ranges>
+
+namespace starflate::huffman::detail {
+
+template <std::ranges::view V>
+class flattened_symbol_bitsize_view
+    : public std::ranges::view_interface<flattened_symbol_bitsize_view<V>>
+{
+  V base_;
+  std::size_t size_;
+
+  using child_view_type = std::remove_cvref_t<std::ranges::range_value_t<V>>;
+
+public:
+  using base_type = V;
+
+  class iterator
+  {
+    const flattened_symbol_bitsize_view* parent_{};
+    std::size_t offset1_{};
+    std::size_t offset2_{};
+
+  public:
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = std::ranges::range_difference_t<child_view_type>;
+    using value_type = std::ranges::range_value_t<child_view_type>;
+    using reference = std::ranges::range_reference_t<child_view_type>;
+    using pointer = void;
+
+    iterator() = default;
```

Comment:
Check if this is still necessary with Clang 17

Reply:
i thought the requirement for default destructibility was removed but guess i was wrong